packages/firebase_remote_config/lib/src/remote_config.dart

```diff
@@ -0,0 +1,232 @@
+part of firebase_remote_config;
+
+/// The entry point for accessing Remote Config.
+///
+/// You can get an instance by calling [RemoteConfig.instance]. Note
+/// [RemoteConfig.instance] is async.
+class RemoteConfig extends ChangeNotifier {
+  static const MethodChannel _channel =
+      const MethodChannel('plugins.flutter.io/firebase_remote_config');
+
+  static const String defaultValueForString = '';
+  static const int defaultValueForInt = 0;
+  static const double defaultValueForDouble = 0.0;
+  static const bool defaultValueForBool = false;
+
+  static const String _fetchFailedThrottledKey = 'FETCH_FAILED_THROTTLED';
+  static const String _lastFetchTimeKey = 'LAST_FETCH_TIME';
+  static const String _lastFetchStatusKey = 'LAST_FETCH_STATUS';
+  static const String _parametersKey = 'PARAMETERS';
+
+  static RemoteConfig _instance;
+
+  Map<String, RemoteConfigValue> _parameters;
+
+  DateTime _lastFetchTime;
+  LastFetchStatus _lastFetchStatus;
+  RemoteConfigSettings _remoteConfigSettings;
+
+  RemoteConfig._();
+
+  DateTime get lastFetchTime => _lastFetchTime;
+  LastFetchStatus get lastFetchStatus => _lastFetchStatus;
+  RemoteConfigSettings get remoteConfigSettings => _remoteConfigSettings;
+
+  /// Gets the instance of RemoteConfig for the default Firebase app.
+  static Future<RemoteConfig> get instance async {
+    if (_instance != null) {
```

Comment:
There's a race condition here where it's possible to have two `instance` objects outstanding if you read RemoteConfig.instance twice in a row before the first `invokeMethod` returns. Consider using a [`Completer`](https://api.dartlang.org/stable/1.24.3/dart-async/Completer-class.html) to avoid this

Reply:
Done